webapp/lib/view.dart

```diff
@@ -1371,6 +1305,81 @@ class ProjectConfigurationView extends BaseView{
   }
 }
 
+abstract class BaseElement {
+  Element get renderElement;
+}
+
+class DropdownElement extends BaseElement {
+  Element _dropdown;
+
+  DropdownElement(Map<String, Function> dropdownItems) {
+    _dropdown = new Element.ul()
+      ..classes.add('add-tag-dropdown');
+    var items = dropdownItems.isEmpty ? ['--None--'] : dropdownItems.keys;
```

Comment:
this isn't adding a function to `dropdownItems`, so the `onClick` below is going to fail. I suggest you create a copy of the map (so you don't edit the original one), and then you add `'--None--'` to that. Something like:
```
var items = new Map.from(dropdownItems);
items.putIfAbsent('--None--', () => (_) {});
```

Reply:
As discussed earlier were retaining this